third_party/skia/platform_view_rtree_unittests.cc

```diff
@@ -0,0 +1,229 @@
+/*
+ * Copyright 2012 Google Inc.
+ *
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+
+#include "flutter/testing/testing.h"
+#include "platform_view_rtree.h"
+
+#include "third_party/skia/include/core/SkCanvas.h"
+#include "third_party/skia/include/core/SkPictureRecorder.h"
+
+namespace flutter {
+namespace testing {
+
+TEST(PlatformViewRTree, NoIntersection) {
+    auto r_tree = sk_make_sp<PlatformViewRTree>();
+    auto rtree_factory = PlatformViewRTreeFactory(r_tree);
+    auto recorder = std::make_unique<SkPictureRecorder>();
+    auto recording_canvas = recorder->beginRecording(SkRect::MakeIWH(1000, 1000), &rtree_factory);
+
+    auto rect_paint = SkPaint();
+    rect_paint.setColor(SkColors::kCyan);
+    rect_paint.setStyle(SkPaint::Style::kFill_Style);
+
+    // If no rect is intersected with the query rect, then the result list is empty.
+    recording_canvas->drawRect(SkRect::MakeLTRB(20, 20, 40, 40), rect_paint);
+    recorder->finishRecordingAsPicture();
+
+    auto hits = r_tree->searchRects(SkRect::MakeLTRB(40, 40, 80, 80));
```

Comment:
Doesn't this test case means the platform views has a single pixel intersection with the drawing? (pixel 40, 40)

Reply:
This part isn't handled by the RTree since it requires iterating over the list again.

Reply:
Not sure I follow, does it mean that due to this rtree implementation we will end up with pixel that should be drawn on top of a platform view but will instead be drawn below?

Reply:
It looks like the bounds are exclusive: https://fiddle.skia.org/c/1f9792323985b4638f8807ffabf3938d

Reply:
To your point, I don't think any pixels will be drawn on the overlay, since the height will be 0 in this case.

Reply:
Not sure I follow, using the same numbers from this test case, lets say we have a red platform view positioned at (40, 40, 80, 80), and on top of it Flutter is drawing a blue rectangle at (20, 20, 40, 40). The expected color of pixel (40, 40) should be blue, but if we won't detect this as an intersection won't we end up painting the blue to the background canvas and having pixel (40, 40) show as red in the final scene?

Reply:
just to be clear, the test is calling `SkRect::MakeLTRB`, which is left, top, right, bottom. In your example, you are assuming, x, y, width, height, right?

Reply:
I'm assuming left top right bottom

Reply:
Anyway if what I'm saying is correct it probably means we have to search for a rect that's wider by 2 pixels and higher than pixels than the actual platform views?

I'm fine with that, though lets be explicit about it in the method's comment

Reply:
In your example, `(40, 40, 80, 80)` platform view and ` (20, 20, 40, 40)` blue rectangle on top should result in no overlay.  The size of this intersection is the equivalent of the rect: `(40, 40, 40, 40)`, which is empty. This is my understanding, but I could be wrong. I can ask the Skia folks to confirm.

Reply:
What should be the color of the pixel at (40,40) in the example above with the red platform view and the blue overlay?

Reply:
The right and bottom values are exclusive. The pixel at (40,40,41,41) is the platform view itself

Reply:
I'm not convinced that a flow platform_view layer that positions a platform view at 0,0, with no transforms will result in the platform view positioned on the screen at (1,1).
Anyway it's a matter of what values we pass in, to my current understanding we need to query for a bigger rect than the actual platform view, but I may very well be missing something. The discussion of what to pass in belongs in the next PR. For this PR lets make sure the comment is explicit about not including the query's edges.

Reply:
resolved offline. `SkRect::MakeLTRB(0,0,1,1)` paints 1 pixel.